lib/pbench/server/api/resources/query_apis/iteration_samples.py

```diff
@@ -0,0 +1,240 @@
+from flask import jsonify
+from http import HTTPStatus
+from logging import Logger
+
+from flask_restful import abort
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.auth import Auth
+from pbench.server.api.resources import (
+    JSON,
+    Schema,
+    Parameter,
+    ParamType,
+    PostprocessError,
+)
+from pbench.server.api.resources.query_apis import CONTEXT, ElasticBase
+from pbench.server.database.models.datasets import Dataset, Metadata, MetadataError
+from pbench.server.database.models.users import User
+from pbench.server.database.models.template import Template, TemplateNotFound
+
+
+class IterationSamples(ElasticBase):
```

Comment:
It occurred to me while I was hiking ("the trees told me") that this pattern of "authorize using a dataset based on a run_id" will be repeated also for TOC and timeseries data, and isn't that different from the publish pattern (albeit with controller/name instead of run_id).

I'm back to thinking this should be "hoisted" into a common place and given a better name and perhaps even a more general structure to accommodate them.

Reply:
Interesting point, the universe is telling us to do that I guess. I believe we could refactor the parent `preprocess` method from `ElasticBase` and do the all the stuff related authorization from dataset there (since it would be common to all)

Reply:
Almost makes me wonder if we might want to do something like moving these classes (plus TOC and timeseries queries) into another sub-module under `query_apis` with this in the `__init__.py` ... like maybe `run_id` ... 😁